### PR TITLE
Implement _save() and _load() counterfactual manager

### PR DESCRIPTION
**Pull request by gaugup**

### CODE REVIEW

File: responsibleai/responsibleai/_managers/counterfactual_manager.py [hunk 1]
```diff
@@ -329,8 +365,88 @@ def name(self):
...
+                    result_file)
+
+    @classmethod
+    def _load(cls, path, model_analysis):
+        """Load the CounterfactualManager from the given path.
+
+        :param path: The directory path to load the CounterfactualManager from.
+        :type path: str
+        :param model_analysis: The loaded parent ModelAnalysis.
+        :type model_analysis: ModelAnalysis
```
Comment (imatiach-msft) [inline responsibleai/responsibleai/_managers/counterfactual_manager.py:398]: is this missing return type?  return/rtype

File: responsibleai/responsibleai/_managers/counterfactual_manager.py [hunk 2]
```diff
@@ -329,8 +365,88 @@ def name(self):
...
+
+            if cf_result['counterfactual_obj'] is not None:
+                counterfactual_config.counterfactual_obj = \
+                    CounterfactualExplanations.from_json(
+                        cf_result['counterfactual_obj'])
+            else:
+                counterfactual_config.counterfactual_obj = None
+            counterfactual_config.has_computation_failed = \
+                cf_result['has_computation_failed']
+            counterfactual_config.failure_reason = cf_result['failure_reason']
```
Comment (imatiach-msft) [inline responsibleai/responsibleai/_managers/counterfactual_manager.py:446]: nit: maybe move some of these repeating strings to be constants at the top of this file

**Review by imatiach-msft — APPROVED** [review]

File: responsibleai/responsibleai/_managers/counterfactual_manager.py (hunk 1 above)
Comment (gaugup) [reply responsibleai/responsibleai/_managers/counterfactual_manager.py:398]: Done

File: responsibleai/responsibleai/_managers/counterfactual_manager.py (hunk 2 above)
Comment (gaugup) [reply responsibleai/responsibleai/_managers/counterfactual_manager.py:446]: Done

**Pull request merged**